### DISCUSSION TITLE
Proxy Protocol Support / --haproxy-clientip

### DISCUSSION THREAD

**Discussion by marsom** [curl/curl#11613] [General]
Hi,

ok I'm probably wrong, I expected the from following curl:

```shell
curl --ipv4 --haproxy-clientip 192.168.1.2 http://localhost:8080
```

to set the client ip in the proxy protocol header to 192.168.1.2. in the proxy protocol naming for me its the source ip not the destination.

but
https://github.com/curl/curl/blob/78d6232f1f326b9ab4d6cdf9582d8fcc2207ffcb/lib/cf-haproxy.c#L91

sets the destination address. 

when i read ie.
- https://developers.cloudflare.com/spectrum/how-to/enable-proxy-protocol/#the-proxy-protocol-v1-header
- https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-proxy-protocol.html
- https://community.f5.com/t5/codeshare/proxy-protocol-initiator/ta-p/280541

i think the client ip = source ip

if I dump the payload from the command obove i get

```text
PROXY TCP4 127.0.0.1 192.168.1.2 62190 8080
GET / HTTP/1.1
Host: localhost:8080
User-Agent: curl/8.2.1
Accept: */*
...
```

**Comment by icing** [comment]
I agree that the implementation looks wrong here. @bagder: wdyt?

**Reply by bagder** [reply]
Indeed. I made  #11619

**Comment by bagder** [comment]
Fixed now.
